src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -185,19 +188,26 @@ public void syncPostBatch(final String topicId, final List<BatchItem> batch) thr
             final boolean isSuccessful = done.await(kafkaSettings.getKafkaSendTimeoutMs(), TimeUnit.MILLISECONDS);
 
             if (!isSuccessful) {
-                failBatch(batch, "timed out");
+                failUnpublished(batch, "timed out");
                 throw new EventPublishingException("Timeout publishing events");
             }
+
+            final boolean atLeastOneFailed = batch.stream()
+                    .anyMatch(item -> item.getResponse().getPublishingStatus() == EventPublishingStatus.FAILED);
+            if (atLeastOneFailed) {
```

Comment:
This corner case would be removed once we have proper timeout configuration in place. Then, there would be no need for "failUnpublished" as well.